src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -122,3 +125,46 @@ def __len__(self):
 
     def __getitem__(self, idx):
         return self.state_rvs[idx]
+
+    def sample(self, locations=None, size=()):
+        """
+        Draw samples from the filtering/smoothing posterior.
+
+
+        If location is None, it returns N joint samples from the list of RVs;
+        if location is one of the locations of the posterior,
+        if not (i.e. if it is a scalar), it returns a sample from the specified location.
+
+        If size is empty, it is a single sample. If not, multiple samples at once.
+        """
+        errormsg = "Sampling not implemented."
+
+        if locations is None:
+            locations = self.locations
+            random_vars = self.state_rvs
+        else:
+            random_vars = _RandomVariableList(
```

Comment:
There's no need to loop over the `locations` here I think, `__call__(iterable)` does essentially the same

Reply:
True. (I wrote this before I decided to put the loop into __call__ )